Philips.CodeAnalysis.SecurityAnalyzers/LicenseAnalyzer.cs

```diff
@@ -1,43 +1,687 @@
+					return foundPath;
+				}
+			}
+
+			var currentDir = Directory.GetCurrentDirectory();
+			var fallbackPath = SearchForAssetsFile(context, currentDir, enableDetailedLogging: false);
+			if (fallbackPath != null)
+			{
+				return fallbackPath;
+			}
+
+			ReportDetailedSearchDiagnostics(context, sourceDirectories, currentDir);
+			return null;
+		}
+
+		private static List<string> GetSourceDirectories(CompilationAnalysisContext context)
+		{
+			return context.Compilation.SyntaxTrees
+				.Where(tree => !string.IsNullOrEmpty(tree.FilePath))
+				.Select(tree => Path.GetDirectoryName(tree.FilePath))
+				.Where(dir => !string.IsNullOrEmpty(dir))
+				.Distinct(StringComparer.OrdinalIgnoreCase)
+				.ToList();
+		}
+
+		private void ReportDetailedSearchDiagnostics(CompilationAnalysisContext context, List<string> sourceDirectories, string currentDir)
+		{
+			ReportDebugDiagnostic(context, $"Could not find {ProjectAssetsFileName}. Starting detailed search diagnostics...");
+			ReportDebugDiagnostic(context, $"Found {sourceDirectories.Count} unique source directories from compilation");
+
+			foreach (var sourceDir in sourceDirectories)
+			{
+				ReportDebugDiagnostic(context, $"Searching from source directory: {sourceDir}");
+				_ = SearchForAssetsFile(context, sourceDir, enableDetailedLogging: true);
+			}
+
+			ReportDebugDiagnostic(context, $"Source directory search failed, falling back to current directory: {currentDir}");
+			_ = SearchForAssetsFile(context, currentDir, enableDetailedLogging: true);
+
+			ReportDebugDiagnostic(context, $"Exhaustive search completed, {ProjectAssetsFileName} not found");
+		}
+
+		private string SearchForAssetsFile(CompilationAnalysisContext context, string startDirectory, bool enableDetailedLogging)
+		{
+			// First, check common locations
+			var foundPath = CheckCommonLocations(context, startDirectory, enableDetailedLogging);
+			if (foundPath != null)
+			{
+				return foundPath;
+			}
+
+			// If not found, search up the directory tree
+			return SearchDirectoryTree(context, startDirectory, enableDetailedLogging);
+		}
+
+		private string CheckCommonLocations(CompilationAnalysisContext context, string startDirectory, bool enableDetailedLogging)
+		{
+			var possiblePaths = new[]
+			{
+				Path.Combine(startDirectory, "obj", ProjectAssetsFileName),
+				Path.Combine(startDirectory, "..", "obj", ProjectAssetsFileName),
+				Path.Combine(startDirectory, "..", "..", "obj", ProjectAssetsFileName)
+			};
+
+			foreach (var path in possiblePaths)
+			{
+				try
+				{
+					var exists = File.Exists(path);
+					if (enableDetailedLogging)
+					{
+						ReportDebugDiagnostic(context, $"Checking path: {path} - Exists: {exists}");
+					}
+
+					if (exists)
+					{
+						return path;
+					}
+				}
+				catch (Exception ex)
+				{
+					if (enableDetailedLogging)
+					{
+						ReportDebugDiagnostic(context, $"Exception checking path {path}: {ex.Message}");
+					}
+				}
+			}
+
+			return null;
+		}
+
```

Comment:
Passing along this warning from sonar: Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed.